packages/contracts-communication/contracts/InterchainClientV1.sol

```diff
@@ -142,15 +142,17 @@ contract InterchainClientV1 is Ownable, IInterchainClientV1 {
         icTx.dbNonce = IInterchainDB(interchainDB).writeEntryWithVerification{value: verificationFees}(
             icTx.dstChainId, icTx.transactionId, srcModules
         );
-        IExecutionService(srcExecutionService).requestExecution({
-            dstChainId: dstChainId,
-            // TODO: this should be encodedTx.length
-            txPayloadSize: message.length,
-            transactionId: transactionId,
-            executionFee: executionFee,
-            options: options
-        });
-        IExecutionFees(executionFees).addExecutionFee{value: executionFee}(dstChainId, transactionId);
+        if (srcExecutionService != address(0)) {
+            IExecutionService(srcExecutionService).requestExecution({
+                dstChainId: dstChainId,
+                // TODO: this should be encodedTx.length
+                txPayloadSize: message.length,
+                transactionId: transactionId,
+                executionFee: executionFee,
+                options: options
+            });
+            IExecutionFees(executionFees).addExecutionFee{value: executionFee}(dstChainId, transactionId);
```

Comment:
There could be a situation now where execution service is not provided, but the execution fee is non-zero. Let's move the addExecutionFee call out of that if statement then?